linker/shared/src/main/scala/org/scalajs/linker/frontend/modulesplitter/ModuleSplitter.scala

```diff
@@ -177,11 +177,16 @@ final class ModuleSplitter private (analyzer: ModuleAnalyzer) {
 }
 
 object ModuleSplitter {
-  def minSplitter(): ModuleSplitter =
-    new ModuleSplitter(new MinModuleAnalyzer())
 
-  def maxSplitter(): ModuleSplitter =
-    new ModuleSplitter(new MaxModuleAnalyzer())
+  def smallestModules(): ModuleSplitter =
+    new ModuleSplitter(new SmallestModulesAnalyzer())
+
+  def fewestModules(): ModuleSplitter =
+    new ModuleSplitter(new FewestModulesAnalyzer())
+
+  def smallModulesFor(packages: List[String]): ModuleSplitter =
+    // TODO: What if a prefix is not valid?
```

Comment:
@sjrd I guess we don't have another choice but to duplicate the validation logic in `ClassName` for a `java.lang.String` in `linkter.interface`?

Reply:
I guess the `// TODO` line should be removed now.